roadmap/README.md

```diff
@@ -0,0 +1,148 @@
+- [`Task` Versioning](https://github.com/tektoncd/pipeline/issues/1839) - Also required
+  for [Catalog](#catalog) success
+- [Alternative Task Implementations](https://github.com/tektoncd/pipeline/issues/215)
+- [Local Execution](https://github.com/tektoncd/pipeline/issues/235)
+- Testing and debugging frameworks
+- Emiting events throughout `Pipeline` execution
+- [Config as code](https://github.com/tektoncd/pipeline/issues/859) - This has some cross
+  over with [a related Triggers issue](https://github.com/tektoncd/triggers/issues/189)
+- [Performant Tekton](https://github.com/tektoncd/pipeline/issues/540) - We should set
+  performance requirements and measure against them
+- [Adding support for other architectures](https://github.com/tektoncd/pipeline/issues/856)
+- [Rich type support for params](https://github.com/tektoncd/pipeline/issues/1393)
+
+#### Task Interfaces and PipelineResources
+
+In alpha Tekton Pipelines, [`PipelineResources`](https://github.com/tektoncd/pipeline/issues/1673)
+were the interface between `Tasks` in a `Pipeline`: they were used for sharing typed
+data and variables between `Tasks`. This is being revisited and the feature has been
+deconstructed into:
+
+* `workspaces` - A way for a `Task` to declare data it needs and provides without
+  needing to know about the underlying mechanism used
+* `Task results` (aka "output params") - A way for a `Task` to provide values as outputs
+  which can be provided to downstream `Tasks`
+
+After adding these features we will once again reconsider the design of `PipelineResources`
+and decide what additional features, if any, we need.
+
+We will also need to decide how to continue to release and maintain the code that
+is used to build the images used by the existing `PipelineResources`:
+https://github.com/tektoncd/pipeline/tree/master/cmd.
+
+### SCM support
+
+SCM support in 2019 was handled by
+[the PullRequest Resource](https://github.com/tektoncd/pipeline/blob/master/docs/resources.md#pull-request-resource).
+However [we are revisting PipelineResources](#task-interfaces-and-pipelineresources),
+and likely this `PipelineResource` will become one or more `Tasks` in
+[the catalog](https://github.com/tektoncd/catalog) instead. Once that happens,
+we will need to make decisions about how to release and maintain the
+supporting code.
+
+* [GitHub App support](https://github.com/tektoncd/triggers/issues/189)
+* Support for more SCM providers
+
+### Results
+
+One of the benefits of [defining specifications around CI/CD](#mission-and-vision)
+is that we can start to create tools around inspecting what is moving through our
+Pipelienes. In 2019 we started some design work around a result storage system for
+Tekton, and we want to make progress on this in 2020:
+
+* Design reporting to results store: https://github.com/tektoncd/pipeline/issues/454
+* Snapshoting data for history + repeatability: https://github.com/tektoncd/pipeline/issues/279
```

Comment:
This should be on a `pipeline` roadmap file :angel:

Reply:
Quick thought: I've been thinking of "results" as a new, totally separate project. e.g. i dont think using tekton pipelines should require using tekton results, kinda like tekton triggers. what do you think @vdemeester ?

Reply:
ah :upside_down_face: then it can be in the main "tekton" roadmap for now, and we'll update as we go :stuck_out_tongue_closed_eyes: